src/extensions/animation/index.js

```diff
@@ -108,89 +93,73 @@ function applyAnimationSettings( extraProps, blockType, attributes ) {
-	 * "original" block is not the current one. Thus, an inexisting
-	 * `originalBlockClientId` prop signals that the block is valid.
-	 *
-	 * @param {Function} WrappedBlockEdit A filtered BlockEdit instance.
-	 * @return {Function} Enhanced component with merged state data props.
-	 */
-	withSelect( ( select ) => {
-		return { selected: select( 'core/block-editor' ).getSelectedBlock(), select };
-	} )
-);
-
-const withAnimationSettings = createHigherOrderComponent( ( BlockListBlock ) => {
-	return enhance( ( { select, ...props } ) => {
-		let wrapperProps = props.wrapperProps;
-
-		const blockName	= select( 'core/block-editor' ).getBlockName( props.rootClientId || props.clientId );
-
-		// Block not supported return unmodified.
-		if ( ! allowedBlocks.some( ( allowedBlock ) => allowedBlock.blockType === blockName && ! allowedBlock.animateChildren ) ) {
-			return <BlockListBlock { ...props } />;
-		}
+/**
+ * React hook function to override the default block element to add wrapper props.
+ *
+ * @function useEditorProps
+ * @param {Object} block        Contains the selected block.
+ * @param {string} blockName    String referencing the selected block.
+ * @param {Object} props        Object with block props.
+ * @param {Object} wrapperProps Object with wrapper props used to extend selected block.
+ * @return {Object} Wrapper props to apply to block.
+ */
+const useEditorProps = ( block, blockName, props, wrapperProps ) => {
+	// Apply animation classes to block wrapper only if the animation attribute has changed.
+	const { animation } = props.attributes;
+	const prevAnimation = useRef();
 
-		if ( ! props.attributes?.animation ) {
-			// If animation is unset ensure that `coblocks-animate` class is unset.
-			if ( props.attributes?.className?.includes( 'coblocks-animate' ) ) {
-				const classList = props.attributes?.className.split( ' ' );
-				props.attributes.className = classList.filter( ( c ) => c !== 'coblocks-animate' ).join( ' ' );
-			}
+	// Hide the Block Toolbar while animations are playing.
+	const [ isAnimating, setIsAnimating ] = useState( false );
 
-			return <BlockListBlock { ...props } />;
+	const blockIsAllowed = allowedBlocks.some( ( allowedBlock ) => allowedBlock.blockType === blockName && ! allowedBlock.animateChildren );
+
+	useEffect( () => {
+		if ( !! blockIsAllowed && !! animation ) {
+			prevAnimation.current = animation;
 		}
+	}, [ animation ] );
 
-		const { animation } = props.attributes;
+	// Block not supported return unmodified.
+	if ( ! blockIsAllowed ) {
+		return wrapperProps;
+	}
 
-		// Apply animation classes to block wrapper only if the animation attribute has changed.
-		const prevAnimation = useRef();
-		useEffect( () => {
-			prevAnimation.current = animation;
-		}, [ animation ] );
+	if ( ! animation ) {
+		// If animation is unset ensure that `coblocks-animate` class is unset.
+		if ( props.attributes?.className?.includes( 'coblocks-animate' ) ) {
```

Comment:
I wonder if we can use classnames utility to put a class to false. Avoiding the verry verbose code below.

Reply:
Things got a little confusing here but I think the primary issue is that the `coblocks-animate` class, after extended, would reside under `attributes.className` as opposed to being regularly dynamically created on the fly. So it was kinda added a single time and therefore does not remove itself from `attributes.className`. This complicated logic was used to force removal of the `blocks-animate` class even when it was part of the `attributes.className` string.

That being said there were a lot of moving parts with this effort. It might be worth revising the logic in a separate PR. 

I think part of the issue with revisiting the code here to re-write is that we are likely not addressing any serious performance issue but rather code quality. While code quality of existing software is important it is not relevant to these changes.